src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -338,14 +347,21 @@ sl_status_t SetWifiConfigurations()
     sl_status_t status = SL_STATUS_OK;
 
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
-    // Setting the listen interval to 0 which will set it to DTIM interval
-    sl_wifi_listen_interval_t sleep_interval = { .listen_interval = 0 };
+    // [sl-only] Set the listen interval to the slow polling interval during association
+    sl_wifi_listen_interval_t sleep_interval = { .listen_interval =
+                                                     chip::ICDConfigurationData::GetInstance().GetSlowPollingInterval().count() };
```

Comment:
I understand what you are trying to do here, but the slow polling interval count will not return 0 right which is needed to poll at DTIM 
This should be done during a subscription is started correct

Reply:
This doesn't exactly do that. To be able to change listen interval after associating with the ap, you need to specify to the AP your max listen interval for it to know how long to buffer message and how long before the AP disconnects you.

This is required for us to be able to change the listen interval at run time between 0 (DTIM) a and the Slow Poll configuration.